File: Right_View_BFS.java

// Time Complexity : O(N)
// Space Complexity : O(N)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No
import java.util.*;

public class Right_View_BFS {
	public List<Integer> rightSideView(TreeNode root) {
		List<Integer> result = new ArrayList<>();
		if (root == null)
			return result;
		Queue<TreeNode> q = new LinkedList<>();
		q.add(root);
		while (!q.isEmpty()) {
			int size = q.size();
			int last = -1;
			for (int i = 0; i < size; i++) {
				TreeNode curr = q.poll();
				last = curr.val;
				if (curr.left != null)
					q.add(curr.left);
				if (curr.right != null)
					q.add(curr.right);
			}
			result.add(last);
		}
		return result;
	}
}

File: right_view_bfs.py
#time-n,space-n
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        q=deque()
        q.append(root)
        res=[]
        while q:
            size=len(q)
            print("s",size)
            for i in range(size):
                
                curr=q.popleft()
                if curr.left:
                    q.append(curr.left)
                if curr.right:
                    q.append(curr.right)
            print(curr.val)    
            res.append(curr.val)
        return res
            